### ISSUE TITLE
Correct accessibility issues with headings

### ISSUE THREAD

**Issue by glowcloud** [swagger-api/swagger-ui#10703]
- skipping h2 - h1 is used for the title and then h3 for operations and schemas. h2 is used only for webhooks in OpenAPI 3.1 (rendered on the same level as tags sections starting with h3) and for nested callbacks in OpenAPI 2.0/3.0 (rendered nested in operations which start with h3). The main sections should be unified to start with the same heading.

- nesting h2 inside of section starting with h3 and h4 - callbacks are nested within operations and start with h2, breaking the order of headings (h1 → h3 → h4 → h2)

<img width="383" height="471" alt="Image" src="https://github.com/user-attachments/assets/70eb9d1a-fe1e-44cc-ade2-fd92971200fe" />

- Parameters and Callbacks heading - these headings are used as buttons but have no indication that the user can switch between the Parameters and Callbacks sections or which one is chosen. It is also not possible to focus them when using tab navigation

- tag heading - the heading is used as a button and can expand/collapse the section without any indication that it does so. We should indicate that it can be interacted with, what it does and what the current state (expanded/collapsed) is.

- unify styling of headings - styling of headings should be the same for different components, in this case it’s  h4 that is different for: cURL, request URL, server response, responses, schemas
